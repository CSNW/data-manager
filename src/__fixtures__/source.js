import Source from '../source';
import { single } from './series';

export default function source() {
  const override = new Source();
  override.fetch = jest.fn(() => Promise.resolve(single()[0].values));

  return override;
}
