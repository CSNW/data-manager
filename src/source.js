export default class Source {
  fetch(path, map) {
    // TODO
  }

  query() {
    // TODO
  }
}
