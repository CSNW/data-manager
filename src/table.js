export default function table(path, map) {
  return (_, source) => {
    return source.fetch(path, map).then(values => ({ values }));
  };
}
