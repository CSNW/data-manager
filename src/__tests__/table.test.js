import table from '../table';
import source from '../__fixtures__/source';

test('should fetch values', () => {
  expect(table('data.csv')([], source())).resolves.toMatchSnapshot();
});
