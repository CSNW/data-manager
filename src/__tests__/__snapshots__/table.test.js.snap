// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch values 1`] = `
Object {
  "values": Array [
    Object {
      "a": 0,
      "b": 10,
    },
    Object {
      "a": 1,
      "b": 8,
    },
    Object {
      "a": 2,
      "b": 6,
    },
    Object {
      "a": 3,
      "b": 4,
    },
    Object {
      "a": 4,
      "b": 2,
    },
    Object {
      "a": 5,
      "b": 0,
    },
  ],
}
`;
